File: python/0187-repeated-dna-sequences.py
class Solution:
    def findRepeatedDnaSequences(self, s: str) -> list[str]:
        result = set()
        previous_sequences = set()
        for i in range(len(s) - 9):
            current = s[i:i+10]
            if current in previous_sequences:
                result.add(current)
            previous_sequences.add(current)
        return list(result)

File: csharp/0187-repeated-dna-sequences.cs
public class Solution
{
    public IList<string> FindRepeatedDnaSequences(string s)
    {
        var seen = new HashSet<string>();
        var result = new HashSet<string>();

        for (var i = 0; i < s.Length - 9; i++)
        {
            var cur = s.Substring(i, 10);
            if (seen.Contains(cur))
                result.Add(cur);
            seen.Add(cur);
        }

        return result.ToList();
    }
}